libraries/octree/src/OctreeEntitiesFileParser.cpp

```diff
@@ -292,11 +292,30 @@ bool OctreeEntitiesFileParser::readEntitiesArray(QVariantList& entitiesArray) {
                     }
                 } else {
                     if (entityObject.contains(key) && entityObject[key].isString()) {
-                        const QString url = entityObject[key].toString();
+                        const QString value = entityObject[key].toString();
 
-                        if (url.startsWith("./") || url.startsWith("../")) {
-                            entityObject[key] = _relativeURL.resolved(url).toString();
+                        if (value.startsWith("./") || value.startsWith("../")) {
+                            // URL value.
+                            entityObject[key] = _relativeURL.resolved(value).toString();
                             isDirty = true;
+                        } else if (value.startsWith("{")) {
+                            // Object with URL values.
+                            auto document = QJsonDocument::fromJson(value.toUtf8());
+                            if (!document.isNull()) {
+                                auto object = document.object();
+                                bool isObjectUpdated = false;
+                                for (const QString& key : object.keys()) {
+                                    auto value = object[key].toString();
+                                    if (value.startsWith("./") || value.startsWith("../")) {
+                                        object[key] = _relativeURL.resolved(value).toString();
+                                        isObjectUpdated = true;
+                                    }
+                                }
+                                if (isObjectUpdated) {
+                                    entityObject[key] = QString(QJsonDocument(object).toJson());
+                                    isDirty = true;
+                                }
+                            }
```

Comment:
if you're interested, this is getting pretty close to being able to handle materialData + userData, I think you'll just need to change it slightly to recurse on the values to handle nested objects

also, I forget, is `serverScripts` just comma separated, or is it in curly brackets?

Reply:
I've added a new issue for userData + materialData: https://github.com/kasenvr/project-athena/issues/700

`serverScripts` is poorly named (it should be `serverScript`): it's just a single URL, so no change.

Reply:
weird!  I could have sworn it used to support multiple scripts

Reply:
I checked the code and as far as I can see, it supports only a single script.

Reply:
if I remember right it can connect to a directory

Reply:
nevermind I am thinking about just using one script that just has a bunch of requires to load more